src/nvim/eval.c

```diff
@@ -21032,6 +21032,23 @@ void ex_oldfiles(exarg_T *eap)
     /* Assume "got_int" was set to truncate the listing. */
     got_int = FALSE;
 
+    // File selection prompt on ":oldfiles!"
+    if (eap->forceit) {
+      quit_more = false;
+      nr = prompt_for_number(false);
+      msg_starthere();
+      if (nr > 0) {
```

Comment:
If `:oldfiles` prints 65 items, using just `<` won't allow you to select the 65th item, presumably because `list_find_str()` indexes `l` at `nr - 1`, not `nr`.